### PR TITLE
feat: asset archive extraction status

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
In this PR I added decompression status update during the decompression process in server and worker.

### CODE REVIEW

File: worker/internal/usecase/gateway/cms.go [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-import "context"
+import (
+	"context"
+
+	"github.com/reearth/reearth-cms/server/pkg/asset"
+)
 
 type CMS interface {
-	NotifyAssetDecompressed(ctx context.Context, assetID string) error
+	NotifyAssetDecompressed(ctx context.Context, assetID string, status *asset.Status) error
```
Comment (rot1024) [inline worker/internal/usecase/gateway/cms.go:10]: Don't import server module in worker module. Do not complicate dependencies easily. Define asset.Status in the worker package

File: server/schemas/asset.graphql [hunk 2]
```diff
@@ -14,6 +14,7 @@ type Asset implements Node {
   thread: Thread
   threadId: ID!
   url: String!
+  status: Status
```
Comment (rot1024) [inline server/schemas/asset.graphql:17]:
```suggestion
  archiveExtractionStatus: Status
```

too abstract name

File: server/schemas/asset.graphql [hunk 3]
```diff
@@ -32,6 +33,13 @@ enum PreviewType {
   UNKNOWN
 }
 
+enum Status {
```
Comment (rot1024) [inline server/schemas/asset.graphql:36]:
```suggestion
enum ArchiveExtractionStatus {
```

too abstract name

File: server/pkg/asset/status_test.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			res, ok := StatusFrom(tc.Name)
+			assert.Equal(t, tc.Expected.TA, res)
+			assert.Equal(t, tc.Expected.Bool, ok)
```
Comment (rot1024) [inline server/pkg/asset/status_test.go:86]: DO NOT USE TABLE DRIVEN TESTS when there are few test cases (at least less than 10 test cases).

Maintaining table-driven tests are more painful than you can imagine. Use table-driven tests only if the number of test cases is greater than the pain.

File: server/internal/usecase/interactor/asset_test.go [hunk 5]
```diff
@@ -648,7 +650,8 @@ func TestAsset_UpdateFiles(t *testing.T) {
 				return mockFs()
 			},
 			assetID: assetID1,
-			want:    asset.New().ID(assetID1).Project(proj.ID()).CreatedByUser(uid).Size(1000).UUID("5130c89f-8f67-4766-b127-49ee6796d464").File(f1).Thread(thid).MustBuild(),
+			status:  sp,
+			want:    asset.New().ID(assetID1).Project(proj.ID()).CreatedByUser(uid).Size(1000).UUID("5130c89f-8f67-4766-b127-49ee6796d464").File(f1).Thread(thid).Status(sp).MustBuild(),
```
Comment (rot1024) [inline server/internal/usecase/interactor/asset_test.go:654]: too long line. Insert line breaks.

File: server/internal/adapter/gql/gqlmodel/convert_asset.go [hunk 6]
```diff
@@ -90,6 +91,28 @@ func ToPreviewType(p *asset.PreviewType) *PreviewType {
 	return &p2
 }
 
+func ToStatus(s *asset.Status) *Status {
```
Comment (rot1024) [inline server/internal/adapter/gql/gqlmodel/convert_asset.go:94]:
```suggestion
func ToArchiveExtractionStatus(s *asset.ArchiveExtractionStatus) *ArchiveExtractionStatus {
```

File: server/pkg/asset/status.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package asset
+
+import (
+	"strings"
+)
+
+type Status string
```
Comment (rot1024) [inline server/pkg/asset/status.go:7]:
```suggestion
type ArchiveExtractionStatus string
```

File: server/pkg/asset/builder.go [hunk 8]
```diff
@@ -103,3 +103,8 @@ func (b *Builder) Thread(th ThreadID) *Builder {
 	b.a.thread = th
 	return b
 }
+
+func (b *Builder) Status(s *Status) *Builder {
```
Comment (rot1024) [inline server/pkg/asset/builder.go:107]:
```suggestion
func (b *Builder) ArchiveExtractionStatus(s *Status) *Builder {
```

File: server/pkg/asset/asset.go [hunk 9]
```diff
@@ -70,6 +71,13 @@ func (a *Asset) UUID() string {
 	return a.uuid
 }
 
+func (a *Asset) Status() *Status {
```
Comment (rot1024) [inline server/pkg/asset/asset.go:74]:
```suggestion
func (a *Asset) ArchiveExtractionStatus() *ArchiveExtractionStatus {
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/internal/adapter/gql/gqlmodel/convert_asset.go (hunk 6 above)
Comment (nourbalaha) [reply server/internal/adapter/gql/gqlmodel/convert_asset.go:94]: Done!

File: server/internal/usecase/interactor/asset_test.go (hunk 5 above)
Comment (nourbalaha) [reply server/internal/usecase/interactor/asset_test.go:654]: Done!

File: server/pkg/asset/asset.go (hunk 9 above)
Comment (nourbalaha) [reply server/pkg/asset/asset.go:74]: Done!

File: server/pkg/asset/builder.go (hunk 8 above)
Comment (nourbalaha) [reply server/pkg/asset/builder.go:107]: Done!

File: server/pkg/asset/status.go (hunk 7 above)
Comment (nourbalaha) [reply server/pkg/asset/status.go:7]: Done!

File: server/schemas/asset.graphql (hunk 3 above)
Comment (nourbalaha) [reply server/schemas/asset.graphql:36]: Done!

File: worker/internal/usecase/gateway/cms.go (hunk 1 above)
Comment (nourbalaha) [reply worker/internal/usecase/gateway/cms.go:10]: Done!

File: server/pkg/asset/status_test.go (hunk 4 above)
Comment (nourbalaha) [reply server/pkg/asset/status_test.go]: Done!

File: worker/pkg/asset/status.go [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+type ArchiveExtractionStatus string
+
+const (
+	ArchiveExtractionStatusPending    ArchiveExtractionStatus = "pending"
+	ArchiveExtractionStatusInProgress ArchiveExtractionStatus = "in_progress"
+	ArchiveExtractionStatusDone       ArchiveExtractionStatus = "done"
+	ArchiveExtractionStatusFailed     ArchiveExtractionStatus = "failed"
+)
+
+func StatusFrom(s string) (ArchiveExtractionStatus, bool) {
```
Comment (rot1024) [inline worker/pkg/asset/status.go:16]:
```suggestion
func ArchiveExtractionStatus From(s string) (ArchiveExtractionStatus, bool) {
```

When you rename a struct, don't forget to rename its associated functions!

File: worker/pkg/asset/status.go [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+	case ArchiveExtractionStatusDone:
+		return ArchiveExtractionStatusDone, true
+	case ArchiveExtractionStatusFailed:
+		return ArchiveExtractionStatusFailed, true
+	default:
+		return ArchiveExtractionStatus(""), false
+	}
+}
+
+func StatusFromRef(s *string) *ArchiveExtractionStatus {
```
Comment (rot1024) [inline worker/pkg/asset/status.go:32]:
```suggestion
func ArchiveExtractionStatusFromRef(s *string) *ArchiveExtractionStatus {
```

File: worker/pkg/asset/status_test.go [hunk 12]
```diff
@@ -0,0 +1,86 @@
+package asset
+
+import (
+	"testing"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStatus_StatusFrom(t *testing.T) {
```
Comment (rot1024) [inline worker/pkg/asset/status_test.go:10]:
```suggestion
func TestArchiveExtractionStatusFrom(t *testing.T) {
```

File: worker/pkg/asset/status_test.go [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+	assert.Equal(t, s, res)
+	assert.True(t, ok)
+
+	s = ArchiveExtractionStatus("")
+	res, ok = StatusFrom("")
+	assert.Equal(t, s, res)
+	assert.False(t, ok)
+}
+
+func TestStatus_StatusFromRef(t *testing.T) {
```
Comment (rot1024) [inline worker/pkg/asset/status_test.go:42]:
```suggestion
func TestArchiveExtractionStatusFromRef(t *testing.T) {
```

**Review by rot1024 — APPROVED** [review]

File: worker/pkg/asset/status_test.go (hunk 12 above)
Comment (nourbalaha) [reply worker/pkg/asset/status_test.go:10]: Done!

File: worker/pkg/asset/status.go (hunk 11 above)
Comment (nourbalaha) [reply worker/pkg/asset/status.go:32]: Done!

File: worker/pkg/asset/status.go (hunk 10 above)
Comment (nourbalaha) [reply worker/pkg/asset/status.go:16]: Done!

File: worker/pkg/asset/status_test.go (hunk 13 above)
Comment (nourbalaha) [reply worker/pkg/asset/status_test.go:42]: Done!

**Pull request merged**